### PR TITLE
Add setting forcelogin to allow redirect to login from homepage.

### PR DESCRIPTION
**Pull request by sumaiyamannan**
Hi,

This setting will check if the user is accessing the site homepage (/index.php) and if the user is not currently logged in, then redirect them to the login page.

Regards,
Sumaiya

### CODE REVIEW

File: settings.php [hunk 1]
```diff
@@ -41,6 +41,12 @@
         '')
     );
 
+    $settings->add(new admin_setting_configcheckbox('local_login/forcelogin',
+        get_string('forcelogin', 'local_login'),
+        get_string('forcelogin_desc', 'local_login'),
+        1)
```
Comment (danmarsden) [inline settings.php]: can we change the default to 0 here - not many people will want this particular setting (I think) - most people will likely use the main Moodle forcelogin setting which forces login for "all pages" not just the homepage like we're doing here.

File: lib.php [hunk 2]
```diff
@@ -76,11 +76,18 @@ function local_login_backgroundimage() {
...
     global $CFG, $FULLME;
+    $forcelogin = get_config('local_login', 'forcelogin');
     $noredirect  = optional_param('noredirect', 0, PARAM_BOOL); // Don't redirect.
     if (stripos($FULLME, $CFG->wwwroot.'/login/index.php') === 0 &&
        !isloggedin() && !empty($noredirect) && !empty($CFG->alternateloginurl)) {
-
         unset($CFG->alternateloginurl);
     }
+
+    if ((stripos($FULLME, $CFG->wwwroot.'/index.php') === 0 || (strlen($FULLME) == strlen($CFG->wwwroot."/"))) &&
```
Comment (danmarsden) [inline lib.php]: I almost merged this as-is, but as the default value in the settings.php page needs to change I wonder if a better solution here might be to use parse_url 

eg something like:
```
$path = parse_url($FULLME, PHP_URL_PATH)
if (($path == "/index.php" || $path == "/") && !isloggedin && $forcelogin == 1)  {
```

Comment (sumaiyamannan) [reply lib.php]: I updated this code to support custom URLs (/moodle/). Please check.

**Comment by sumaiyamannan** [comment]
Hi @danmarsden ,
I have fixed the login to work with index.php.
I have also cherry-picked nmet commit. Please check and merge.

Regards,
Sumaiya

**Pull request merged**